### PR TITLE
Feature/backup attention screen

### PR DESCRIPTION
**Pull request by svojsu**
The Backup Attention Screen is used to warn users about possible bad practices in passphrase and backup management.

Simulator Screenshot - iPhone 15 Pro Max - 2024-05-15 at 19 50 57

### CODE REVIEW

File: novawallet/Common/View/ScrollableContainerView/ScrollableContainerView.swift [hunk 1]
```diff
@@ -22,6 +22,12 @@ final class ScrollableContainerView: UIView {
         }
     }
 
+    var scrollContentBottomOffset: CGFloat = 0 {
```
Comment (ERussel) [inline novawallet/Common/View/ScrollableContainerView/ScrollableContainerView.swift:25]: We usually use stackView.layoutMargins to setup the insets directly in the layout

File: novawallet/Modules/BackupAttention/BackupAttentionViewController.swift [hunk 2]
```diff
@@ -1,13 +1,26 @@
...
+import SoraFoundation
 
-final class BackupAttentionViewController: UIViewController {
+final class BackupAttentionViewController: UIViewController, ViewHolder {
     typealias RootViewType = BackupAttentionViewLayout
 
     let presenter: BackupAttentionPresenterProtocol
 
-    init(presenter: BackupAttentionPresenterProtocol) {
+    private var appearanceAnimator: ViewAnimatorProtocol?
```
Comment (ERussel) [inline novawallet/Modules/BackupAttention/BackupAttentionViewController.swift:10]: We don't use animations to switch between button's enabled/disabled state

File: novawallet/Modules/BackupAttention/BackupAttentionViewLayout.swift [hunk 3]
```diff
@@ -1,6 +1,34 @@
 import UIKit
+import SoraUI
 
 final class BackupAttentionViewLayout: UIView {
+    var checkBoxScrollableView = CheckBoxIconDetailsScrollableView()
+
+    var agreeButton: UIControl?
+
+    var blurredBottomView: OverlayBlurBackgroundView = .create { view in
```
Comment (ERussel) [inline novawallet/Modules/BackupAttention/BackupAttentionViewLayout.swift:9]: Please, check StartStakingInfoViewLayout for the reference. It uses BlurBackgroundView to setup similar footer

File: novawallet/Modules/BackupAttention/BackupAttentionViewLayout.swift [hunk 4]
```diff
@@ -10,4 +38,126 @@ final class BackupAttentionViewLayout: UIView {
...
+    func setupAgreeButton(with button: UIControl) {
+        guard let agreeButton else {
+            showButtonWithAnimation(button)
+            return
+        }
+
+        changeButtonWithAnimation(button)
+    }
+
+    func changeButtonWithAnimation(_ button: UIControl) {
```
Comment (ERussel) [inline novawallet/Modules/BackupAttention/BackupAttentionViewLayout.swift:130]: We don't need an animation to switch between button's state. Let's use static button and change its title and style to switch between states. For styling we usually use applyEnabledStyle/applyDisableStyle

File: novawallet/Modules/BackupAttention/BackupAttentionViewLayout.swift [hunk 5]
```diff
@@ -1,6 +1,34 @@
 import UIKit
+import SoraUI
 
 final class BackupAttentionViewLayout: UIView {
```
Comment (ERussel) [inline novawallet/Modules/BackupAttention/BackupAttentionViewLayout.swift:4]: Let's inherit here from ScrollableContainerLayoutView and use stackView.layoutMargins to control bottom offset like it is done in StartStakingInfoViewLayout

File: novawallet/ru.lproj/Localizable.strings [hunk 6]
```diff
@@ -1589,3 +1589,12 @@
...
 "backup.select.wallet.title" = "Выберите кошелек для бэкапа";
+"backup.attention.title" = "Не сообщайте свою секретную фразу!";
+"backup.attention.description" = "Пожалуйста, внимательно прочитайте следующее перед просмотром резервной копии";
+"backup.attention.passphrase.description" = "Эта секретная фраза дает вам полный и постоянный доступ ко всем подключенным кошелькам и находящимся в них средствам.\n%@";
+"backup.attention.passphrase.description.highlighted" = "НЕ ДЕЛИТЕСЬ ЕЙ.";
+"backup.attention.passphrase.write.description" = "Не вводите свою секретную фразу ни в какие формы и сайты.\n%@";
+"backup.attention.passphrase.write.description.highlighted" = "СРЕДСТВА МОГУТ БЫТЬ ПОТЕРЯНЫ НАВСЕГДА.";
+"backup.attention.passphrase.support.description" = "Служба поддержки или администраторы никогда и ни при каких обстоятельствах не будут запрашивать вашу секретную фразу.\n%@";
+"backup.attention.passphrase.support.description.highlighted" = "ОСТЕРЕГАЙТЕСЬ ПОДРАЖАТЕЛЕЙ.";
+"backup.attention.aggree.button.title" = "Просмотрите и примите, чтобы продолжить";
```
Comment (ERussel) [inline novawallet/ru.lproj/Localizable.strings:1600]: Le't just leave "Проверьте и примите"

File: novawallet/Modules/BackupAttention/View/BackupAttentionTableTitleView.swift [hunk 7]
```diff
@@ -0,0 +1,72 @@
+import UIKit
+import SoraUI
+
+final class BackupAttentionTableTitleView: GenericPairValueView<
+    UIView,
+    GenericPairValueView<UILabel, UILabel>
+> {
+    var warningIconContainerView: UIView { fView }
```
Comment (ERussel) [inline novawallet/Modules/BackupAttention/View/BackupAttentionTableTitleView.swift:8]: Can't simplified by using DAppIconView

File: novawallet/Modules/BackupAttention/BackupAttentionViewLayout.swift [hunk 8]
```diff
@@ -1,158 +1,117 @@
...
-        view.cornerCut = [.topLeft, .topRight]
+    let footer: BlurBackgroundView = .create {
+        $0.sideLength = UIConstants.footerCornerRadius
+        $0.cornerCut = [.topLeft, .topRight]
     }
 
-    private var viewModel: Model?
-    private var appearanceAnimator: ViewAnimatorProtocol?
-    private var disappearanceAnimator: ViewAnimatorProtocol?
+    let agreeButton: TriangularedButton = {
```
Comment (ERussel) [inline novawallet/Modules/BackupAttention/BackupAttentionViewLayout.swift:19]: more compact way is ```= .create { button in button.applyDefaultStyle() }```

**Review by ERussel — APPROVED** [review]

File: novawallet/Common/Extension/UIKit/TriangularedBlurButton+Style.swift [hunk 9]
```diff
@@ -0,0 +1,15 @@
+import Foundation
+
+extension TriangularedBlurButton {
```
Comment (ERussel) [inline novawallet/Common/Extension/UIKit/TriangularedBlurButton+Style.swift:3]: Do we still need this?

Comment (svojsu) [reply novawallet/Common/Extension/UIKit/TriangularedBlurButton+Style.swift:3]: no, we don't

**Review by ERussel — APPROVED** [review]

**Pull request merged**